konfig

```diff
@@ -84,28 +85,49 @@ merge() {
 }
 
 import_ctx() {
+    declare -a tmpcfgs=()
     local tmpcfg
+    local tmpinputcfg=""
     local out=""
     local arg=""
-    if [[ "$1" == '-p' || "$1" == '--preserve-structure' ]]; then
-      arg="$1"
-      shift
-    fi
-    if [[ "$1" = '--save' || "$1" = '-s' ]]; then
-      out="${XDG_CACHE_HOME:-$HOME/.kube}/config"
-      shift
-    elif [[ "$1" =~ ^-(.*) ]]; then
-      error "unrecognized flag \"$1\""
-    fi
+    for OPT in "$@"; do
+      case $OPT in
+        -p | --preserve-structure)
+          arg="$1"
+          shift
+          ;;
+        -s | --save)
+          out="${XDG_CACHE_HOME:-$HOME/.kube}/config"
+          shift
+          ;;
+        -i | --stdin)
+          if [[ -p /dev/stdin ]]; then
+            tmpinputcfg=$(mktemp konfig_input_XXXXXX)
+            TMPFILES+=( "$tmpinputcfg" )
+            cat - > "$tmpinputcfg"
+            shift
+          fi
+          ;;
+        -*)
+          error "unrecognized flag \"$1\""
+          ;;
+      esac
+    done
     tmpcfg=$(mktemp konfig_XXXXXX)
     TMPFILES+=( "$tmpcfg" )
     $KUBECTL config view --raw > "$tmpcfg"
 
+    # Add $HOME/.kube first to keep current context of $HOME/.kube
+    tmpcfgs+=( "$tmpcfg")
+    if [[ -n "$tmpinputcfg" ]]; then
+      tmpcfgs+=( "$tmpinputcfg" )
+    fi
```

Comment:
IIUC, you could move the 
```
tmpcfg=$(mktemp konfig_XXXXXX)
TMPFILES+=( "$tmpcfg" )
```
lines before the loop, to add to the array in the loop. Then you don't have to use the conditional if block `if [[ -n "$tmpinputcfg" ]];` below.

Does that work?

Reply:
Yes, it works. I reflected your review to the commit.  Thank you so much for bearing with me.